### PR TITLE
Dissecting Biddle Traitors: Toned down traitor flavor + antag info tgui

### PR DESCRIPTION
**Pull request by tralezab**
## what 
<details>
  <summary>Examples of Antag Tgui done so far</summary>
  
  #### Traitor Tgui (subject to change)
  image

  #### Morph Tgui as an example of a simple one
  image

  #### Nightmare Tgui
  image

  #### Brainwashed Tgui
  image

</details>

also, traitor flavor is in as it made up a fair chunk of the UI and removing it was far less convenient than just including it, specially since it has no gameplay purpose and is just an antag mind jogger

## why

needed in #60302

okay but really storing antag greetings and important info in chat which them scrolls and deletes is a bad idea, having to make the traitor go to notes for any reason ever is an even worse idea.

## Changelog
:cl:
add: traitors now get some extra flavor. doesn't mean anything mechanically or ruleswise right now, but maybe it will inspire some more creative traitoring?
add: a few antags now have a tgui menu instead of a greet. spiffy! they include antag info you need.
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
For more information read up on the original pr these features were in #59442

**Comment by tralezab** [comment]
ugh i can already hear the maintainers whining in the distance 1 sec

**Comment by MrDoomBringer** [comment]
There needs to be a clear note that the 'Employer' part of the traitor-panel is fluff and doesn't need to be followed.

Something straightforward like "Roleplay suggestion" written in small text above the alligences/employer thoughts section would work great.

**Comment by AlinaStarkova** [comment]
> 
> 
> There needs to be a clear note that the 'Employer' part of the traitor-panel is fluff and doesn't need to be followed.
> 
> Something straightforward like "Roleplay suggestion" written in small text above the alligences/employer thoughts section would work great.

a note shouldnt be nesscary, nearly every antag already has fluff text that doesnt need to be followed. its even in TG policy that antag status doesnt have set requirements you need to follow (unless your on manuel). if someone reads new fluff text and thinks its a requirement thats moreso on them

**Comment by tralezab** [comment]
> There needs to be a clear note that the 'Employer' part of the traitor-panel is fluff and doesn't need to be followed.
> 
> Something straightforward like "Roleplay suggestion" written in small text above the alligences/employer thoughts section would work great.

sounds good, will add



> a note shouldnt be nesscary, nearly every antag already has fluff text that doesnt need to be followed. its even in TG policy that antag status doesnt have set requirements you need to follow (unless your on manuel). if someone reads new fluff text and thinks its a requirement thats moreso on them

i dunno, this one is pretty in depth

**Comment by MrDoomBringer** [comment]
@AlinaStarkova 

> nearly every antag already has fluff text that doesnt need to be followed.

This is the only antag flavor-text that implies limitations to your role.

> if someone reads new fluff text and thinks its a requirement thats moreso on them

There is absolutely no reason a new player would read these notes and think "it's optional" - at best they would read Rule 4, get confused about the apparent contradiction, then have to ask IC/OOC/Admins about it. Player retention for this game is god-awful as is, please don't try to intentionally make it worse.

**Comment by AlinaStarkova** [comment]
> 
> 
> @AlinaStarkova
> 
> > nearly every antag already has fluff text that doesnt need to be followed.
> 
> This is the only antag flavor-text that implies limitations to what you are and aren't allowed to do.

it objectively isnt, contractor implies your meant to kidnap: you dont have to, nukies implies your meant to nuke the station: you dont have to (you can just murderbone), nightmare implies your meant to kill people in the dark of maint: you dont have to. ashwalkers implies you cant invade the station: you can. sentient virus implies you have to infect everyone: you dont have to. cult and rev implies you have to convert everyone and kill X: you dont have to. heretic implies you have to sacrafice people: you dont have to

**Comment by tralezab** [comment]
eh honestly i'm just gonna remove the allegiances section until future prospects

**Review by stylemistake — APPROVED** [review]
This is fucking beautiful.

My only wish if this was a single file, or a tightly packed group of files (e.g. with the same prefix).

**Comment by MrDoomBringer** [comment]
@AlinaStarkova 

>it objectively isnt

I think misworded my argument, sorry for that, was a bit late. Should've said this is the only antag fluff that directly tells players they aren't allowed to do certain things. Nearly every other instance of this is a policy rule that you must follow, for example, revs gets the flavor text "Do not harm your fellow freedom fighters."

You're still right in saying that existing objectives don't seem optional, and that's an issue that we should fix, rather than exacerbate. 

Anyways, looks like the allegiances section is being taken out so for now this is a moot point

**Comment by tralezab** [comment]
i woke up and im in a better mood so im once again keeping the employer section and adding a "THIS IS FLAVOR" warning

**Comment by MrStonedOne** [comment]
So what notification do traitors actually get that they have antag now?

**Review by MrsTonedOne — CHANGES_REQUESTED** [review]
Considering that somebody can get banned if they don't see some of these windows:

I can't support removing the alert from chat. You can have most the details in this, but tgui windows can't be relied upon to open up and they contain info that admins will assume they have seen when deciding on punishments for not following things like the mind control.

**Comment by tralezab** [comment]
Will send in chat a message about using the antag tgui panel

**Review by MrsTonedOne — APPROVED** [review]

File: code/game/objects/items/implants/implantuplink.dm [hunk 1]
```diff
@@ -11,7 +11,8 @@
 
 /obj/item/implant/uplink/Initialize(mapload, owner, uplink_flag)
 	. = ..()
-	AddComponent(/datum/component/uplink, _owner = owner, _lockable = TRUE, _enabled = FALSE, uplink_flag = uplink_flag, starting_tc = starting_tc)
+	var/datum/component/uplink/new_uplink = AddComponent(/datum/component/uplink, owner, TRUE, FALSE, null, starting_tc)
```
Comment (Mothblocks) [inline code/game/objects/items/implants/implantuplink.dm:14]: Why did you remove the named parameters from this? It's totally unreadable now.

File: code/modules/antagonists/_common/antag_datum.dm [hunk 2]
```diff
@@ -133,8 +140,15 @@ GLOBAL_LIST_EMPTY(antagonists)
 		CRASH("[src] ran on_gain() without a mind")
 	if(!owner.current)
 		CRASH("[src] ran on_gain() on a mind without a mob")
+	if(ui_name)//in the future, this should entirely replace greet.
+		info_button = new(owner.current, src)
+		info_button.Grant(owner.current)
 	if(!silent)
 		greet()
+		if(ui_name)
+			to_chat(owner.current, span_big("You are a [src]."))
```
Comment (Mothblocks) [inline code/modules/antagonists/_common/antag_datum.dm:149]:
```suggestion
			to_chat(owner.current, span_big("You are \a [src]."))
```

Otherwise--you are a abductor.

File: code/modules/antagonists/_common/antag_datum.dm [hunk 3]
```diff
@@ -360,3 +376,48 @@ GLOBAL_LIST_EMPTY(antagonists)
...
+	if(!ui)
+		ui = new(user, src, ui_name, name)
+		ui.open()
+
+/datum/antagonist/ui_state(mob/user)
+	return GLOB.always_state
+
+///generic helper to send objectives as data through tgui. supports smart objectives too!
+/datum/antagonist/proc/get_objectives()
+	var/objective_count = 1
```
Comment (Mothblocks) [inline code/modules/antagonists/_common/antag_datum.dm:393]: Why does this start at 1?

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/_common/antag_datum.dm (hunk 3 above)
Comment (tralezab) [reply code/modules/antagonists/_common/antag_datum.dm:393]: I find that since lists start at 1 in byond it helps to start indexing like that too
